src/flake8_to_ruff/black.rs

```diff
@@ -15,16 +17,6 @@ pub struct Black {
     pub target_version: Option<Vec<PythonVersion>>,
 }
 
-#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
-struct Tools {
-    black: Option<Black>,
-}
-
-#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
-struct Pyproject {
-    tool: Option<Tools>,
-}
-
 pub fn parse_black_options<P: AsRef<Path>>(path: P) -> Result<Option<Black>> {
```

Comment:
I think we should have one `parse` function that returns `ExternalConfig`. Right now, we're parsing the TOML twice into `Pyproject`, then extracting `tool.black` and `tool.isort` respectively, then merging them back into a single struct.

Reply:
(Or, one parse function that returns `Pyproject` to `main.rs`, then create `ExternalConfig` from `Pyproject`.)

Reply:
Ah yep that makes sense.

`Tools` has to own `black`/`isort` presumably for `serde` (de)serialization. I'm trying to do something like:

```rust
pub fn parse<'a, P: AsRef<Path>>(path: P) -> Result<ExternalConfig<'a>> {
    let contents = std::fs::read_to_string(path)?;
    let pyproject = toml_edit::easy::from_str::<Pyproject>(&contents)?;
    Ok(pyproject
        .tool
        .map(|tool| ExternalConfig {
            black: tool.black.as_ref(),
            isort: tool.isort.as_ref(),
        })
        .unwrap_or_default())
}
```

but the `tool.{black,isort}.as_ref()` lines complain because we're referencing `tool.{black,isort}` which is not owned by the current function. Any ideas? :)

Reply:
The signature doesn't make sense:

```rs
pub fn parse<'a, P: AsRef<Path>>(path: P) -> Result<ExternalConfig<'a>> {
```

to support such a signature `ExternalConfig` and everything contained will have to be owned.

Reply:
Sorry @not-my-profile, I've changed `flake8_to_ruff::parse` to return `Pyproject` instead, which circumvents the above issue.

The current issue is mapping out of `Tools` (where `black`/`isort` are owned) into `ExternalConfig` (where `black`/`isort` are references).

Reply:
This seems to work, but is there a preferred (more idiomatic) way?

```rust
let mut external_config = ExternalConfig::default();
let pyproject = cli.pyproject.map(flake8_to_ruff::parse).transpose()?;
if let Some(pyproject) = &pyproject {
    if let Some(tool) = &pyproject.tool {
        external_config = ExternalConfig {
            black: tool.black.as_ref(),
            isort: tool.isort.as_ref(),
        };
    }
}
```

Reply:
Implemented suggestions here @charliermarsh: https://github.com/charliermarsh/ruff/pull/2082/commits/4127eb451877e3a687b4c74103dce99f11ceca69 👍🏼

Reply:
Awesome -- will review in a bit.